### DISCUSSION TITLE
RoboticsBackend VNC View Is Slow, and no direct GPU access error

### DISCUSSION THREAD

**Discussion by obamwonyi** [JdeRobot/RoboticsAcademy#2688] [General]
Hi, I have successfully set up RoboticsAcademyBackend locally but the VNC display of the world is very blurry and slow, is there a way I can implement it in gazebo directly, as gazebo runs fine on my system same with ros2, also I got the error below when I ran the docker container 
![Screenshot from 2024-08-07 01-24-09](https://github.com/user-attachments/assets/9f8bdcb8-e819-4d10-a2f2-203be2164abd)

**Comment by jmplaza** [comment]
Hi @obamwonyi,

no, the Gazebo should run inside the RoboticsBackend. Which release of RoboticsBackend are you using? Which host operating system? Which GPU do you have at your computer (none/nVidia/integrated Intel...? 

There are several ways to launch the RoboticsBackend in order to let it use the available GPU, which will speed up the execution of Gazebo. There are all documented at [RoboticsAcademy User Guide](https://jderobot.github.io/RoboticsAcademy/user_guide/). Anyway connecting with the GPU inside the RoboticsBackend docker images is still tricky on some cases...

**Comment by obamwonyi** [comment]
#### Robotics-Backend Release 
> jderobot/robotics-backend  4.6.1 

####  Operating System
> Distributor ID:	Ubuntu
>Description:	Ubuntu 22.04.4 LTS
>Release:	22.04
>Codename:	jammy

#### GPU
> Advanced Micro Devices, Inc. [AMD/ATI] Richland [Radeon HD 8610G]

#### GPU Size
> Video memory: 768MB
>Dedicated video memory: 768 MB
>Currently available dedicated video memory: 767 MB


Please note that, when I run Gazebo and Ros2 (Humble) on my system without a container, they run fast, I did not try any of the GPU acceleration guides in the picture below, as there were none for AMD GPUs.

![Screenshot from 2024-08-07 23-16-13](https://github.com/user-attachments/assets/dd576e06-7272-4c57-b64a-25e4c4a3ca12)

**Comment by jmplaza** [comment]
Uhm, you have a configuration (the AMD GPU) we haven't tested in depth. Try with the latest RoboticsBackend (4.6.5) [https://hub.docker.com/r/jderobot/robotics-backend/tags](https://hub.docker.com/r/jderobot/robotics-backend/tags) , although I don't expect any different behavior.

Have you launched the RoboticsBackend with the "only CPU" way?
`docker run --rm -it -p 7164:7164 -p 6080:6080 -p 1108:1108 -p 7163:7163 jderobot/robotics-backend:4.6.5`
It should run fine on your machine. Anyway try also with the "Intel GPU" launching.

There are two performance indicators: Gazebo Real Time Factor (RTF) and Gazebo FPS. You can see them at the Gazebo viewer widget. What are  the numbers with your "only CPU" launching of the RoboticsBackend?

[Direct Rendering Infrastructure (DRI)](https://en.wikipedia.org/wiki/Direct_Rendering_Infrastructure) is the standard way in linux to provide GPU access to applications. Could you please share the output of these commands on your local Ubuntu OS? 
`lshw -class display` and` ls -altr /dev/dri`. Several problems are reported with AMD GPUs and Linux (for instance: 
https://community.amd.com/t5/blender-discussions/amd-drivers-for-ubuntu-24-04-amd-gpu-pro-for-latest-linux-kernel/m-p/688704 ), maybe one of them is preventing the use of your GPUs from the RoboticsBackend Docker Image...

**Comment by obamwonyi** [comment]
Ok, I will download it when I have enough data. Thanks for your response.

**Comment by obamwonyi** [comment]
#### Output of,  *lshw -class display*
> lshw -class display
  *-display                 
       description: VGA compatible controller
       product: Richland [Radeon HD 8610G]
       vendor: Advanced Micro Devices, Inc. [AMD/ATI]
       physical id: 1
       bus info: pci@0000:00:01.0
       logical name: /dev/fb0
       version: 00
       width: 32 bits
       clock: 33MHz
       capabilities: pm pciexpress msi vga_controller bus_master cap_list rom fb
       configuration: depth=32 driver=radeon latency=0 resolution=1366,768
       resources: irq:35 memory:e0000000-efffffff ioport:4000(size=256) memory:f0400000-f043ffff memory:c0000-dffff

#### Output of,  *ls -altr /dev/dri*
>ls -altr /dev/dri
total 0
crw-rw----+  1 root render 226, 128 Aug 10 15:40 renderD128
drwxr-xr-x   3 root root        100 Aug 10 15:40 .
crw-rw----+  1 root video  226,   0 Aug 10 15:40 card0
drwxr-xr-x   2 root root         80 Aug 10 15:40 by-path
drwxr-xr-x  22 root root       5400 Aug 10 16:00 ..
